File: 18-nucleotide-count/dna.rb
class DNA

  attr_reader :string

  def initialize(string)
    @string = string
  end

  def count(string)
    0
  end

  def process
    up = @string.upcase
    up.split("")
  end

  def nucleotide_counts
    process.collect do |letter|
      nucleotide_symbols.key(letter)
    end
  end

  def empty_nuc
    {'A' => 0, 'T' => 0, 'G' => 0, 'C' => 0}
  end

  def nucleotide_symbols
    {'A' => 6, 'T' => 6, 'G' => 5, 'C' => 4}
  end

end

File: 18-nucleotide-count/dna.js
DNA = function(dna) {
	this.count = function(input) {
		var string = dna;

		if (string === "") {
			return 0;
		} else {
			var r = new RegExp(input, 'g');
			var count = string.match(r);
			var countLength = count.length;
			return countLength;
		}
	};

	this.nucleotideCounts = {
		'A' : 0, 'T' : 0, 'C' : 0, 'G' : 0
	}

	// total counts
	for (var i = 0; i < dna.length; i++) {
		var currentNucleotide = dna[i];
		this.nucleotideCounts[currentNucleotide]++;
	};

};
